featurebyte/service/feature_preview.py

```diff
@@ -73,11 +93,48 @@ def _update_point_in_time_if_needed(
 
         Raises
         ------
+        LimitExceededError
+            raised if the observation table has more than 50 rows
         MissingPointInTimeColumnError
             raised if the point in time column is not provided in the dictionary for a time based feature
         """
+        # Validate the observation_table_id
+        if preview_observation_set.observation_table_id is not None:
+            observation_table = await self.observation_table_service.get_document(
+                document_id=preview_observation_set.observation_table_id
+            )
+            if observation_table.num_rows > 100:
+                raise LimitExceededError("Observation table must have 100 rows or less")
```

Comment:
Should the limit be 50 to be consistent with the DataFrame case? Do we raise the same exception when validating DataFrame observation set?

Reply:
Sure will change to 50 to be consistent